static/js/realm_icon.js

```diff
@@ -0,0 +1,34 @@
+var realm_icon = (function () {
+
+    var exports = {};
+
+    exports.build_realm_icon_widget = function (upload_function) {
+        var get_file_input = function () {
+            return $('#realm_icon_file_input').expectOne();
+        };
+
+        if (page_params.realm_icon_source === 'G') {
+            $("#realm_icon_delete_button").hide();
+        }
```

Comment:
This should probably have a `.show()` here as well if present.  Maybe worth refactoring into a separate function as a follow-up.